### DISCUSSION TITLE
German Umlaute just like in US International Alt Gr (Linux, Windows)

### DISCUSSION THREAD

**Discussion by SeoliteQ** [pqrs-org/Karabiner-Elements#4078] [Show and tell]
Just switched from many years of Linux and Windows to macos and wanted to set up German umlaute without installing a separate keyboard layout in macos (tried that first, but some apps started crashing every now and then, hence here I am).

Found the "German Umlaut" keymap on karabiner's complex modification rules and decided to modify it as it uses a, o and u as the initial key while I am used to q, p and y being used for this.

Code is below for anyone wanting to stick to the linux/windows behavior to reach ä, ö and ü.

{
    "description": "Change option + q/p/y to ä/ö/ü",
    "manipulators": [
        {
            "from": {
                "key_code": "q",
                "modifiers": {
                    "mandatory": ["option"],
                    "optional": ["caps_lock"]
                }
            },
            "to": [
                {
                    "key_code": "u",
                    "modifiers": ["left_option"]
                },
                { "key_code": "a" },
                { "key_code": "vk_none" }
            ],
            "type": "basic"
        },
        {
            "from": {
                "key_code": "q",
                "modifiers": { "mandatory": ["option", "shift"] }
            },
            "to": [
                {
                    "key_code": "u",
                    "modifiers": ["left_option"]
                },
                {
                    "key_code": "a",
                    "modifiers": ["left_shift"]
                },
                { "key_code": "vk_none" }
            ],
            "type": "basic"
        },
        {
            "from": {
                "key_code": "p",
                "modifiers": {
                    "mandatory": ["option"],
                    "optional": ["caps_lock"]
                }
            },
            "to": [
                {
                    "key_code": "u",
                    "modifiers": ["left_option"]
                },
                { "key_code": "o" },
                { "key_code": "vk_none" }
            ],
            "type": "basic"
        },
        {
            "from": {
                "key_code": "p",
                "modifiers": { "mandatory": ["option", "shift"] }
            },
            "to": [
                {
                    "key_code": "u",
                    "modifiers": ["left_option"]
                },
                {
                    "key_code": "o",
                    "modifiers": ["left_shift"]
                },
                { "key_code": "vk_none" }
            ],
            "type": "basic"
        },
        {
            "from": {
                "key_code": "y",
                "modifiers": {
                    "mandatory": ["option"],
                    "optional": ["caps_lock"]
                }
            },
            "to": [
                {
                    "key_code": "u",
                    "modifiers": ["left_option"]
                },
                { "key_code": "u" },
                { "key_code": "vk_none" }
            ],
            "type": "basic"
        },
        {
            "from": {
                "key_code": "y",
                "modifiers": { "mandatory": ["option", "shift"] }
            },
            "to": [
                {
                    "key_code": "u",
                    "modifiers": ["left_option"]
                },
                {
                    "key_code": "u",
                    "modifiers": ["left_shift"]
                },
                { "key_code": "vk_none" }
            ],
            "type": "basic"
        }
    ]
}

**Comment by Dirtinger** [comment]
Thanks a lot! Made the switch to macOS and this was a dealbreaker. Works great.
